identity-account-storage/src/storage/traits.rs

```diff
@@ -118,6 +120,31 @@ pub trait Storage: storage_sub_trait::StorageSendSyncMaybe + Debug {
   /// Returns `true` if a key exists at the specified `location`.
   async fn key_exists(&self, did: &IotaDID, location: &KeyLocation) -> Result<bool>;
 
+  /// Encrypts the given `data` with the specified `algorithm`
+  ///
+  /// Diffie-Helman key exchange will be performed in case an [`KeyType::X25519`] is given.
+  async fn encrypt_data(
+    &self,
+    did: &IotaDID,
+    data: Vec<u8>,
+    associated_data: Vec<u8>,
+    algorithm: &EncryptionOptions,
+    private_key: &KeyLocation,
+    public_key: PublicKey,
+  ) -> Result<EncryptedData>;
+
+  /// Decrypts the given `data` with the specified `algorithm`
+  ///
+  /// Diffie-Helman key exchange will be performed in case an [`KeyType::X25519`] is given.
+  async fn decrypt_data(
```

Comment:
Again please write a bit more documentation here. What could the different parameters indicate/ be used for.